### ISSUE TITLE
kartik\file\FileInput and kartik\base\TranslationTrait define the same property ($i18n) in the composition of kartik\file\FileInput. However, the definition differs and is considered incompatible. Class was composed

### ISSUE THREAD

**Issue by tholag** [kartik-v/bootstrap-fileinput#1295]
Hello, 

I get this error:

`kartik\base\InputWidget and kartik\base\TranslationTrait define the same property ($i18n) in the composition of kartik\file\FileInput. However, the definition differs and is considered incompatible. Class was composed` when using the `kartik\file\FileInput`-Widget

I tried updating all Packages from the "kartijk-v/*"-branch, but it didn't solve the problem (`yii2-krajee-base` is `1.9.4` and `yii2-widget-fileinput` is `v1.0.7` and `kartik-v/bootstrap-fileinput` is `v4.5.0`).

Any hints how to get this problem solved?

Thanks a lot!

Thorsten

## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of bootstrap-fileinput.
- [x] This is not an usage question. I confirm having read the plugin [documentation](http://plugins.krajee.com/file-input) and [demos](http://plugins.krajee.com/file-input/demo).
- [ x This is not a general programming / coding question. (Those should be directed to the [webtips Q & A forum](http://webtips.krajee.com/questions)).
- [x] I have attempted to find the simplest possible steps to reproduce the issue.
- [x] I have included a failing test as a pull request (Optional).
## Environment

Browsers

- [x] Google Chrome
- [x] Mozilla Firefox
- [x] Internet Explorer
- [x] Safari

Operating System

- [ ] Windows
- [ ] Mac OS X
- [x] Linux
- [ ] Mobile

Libraries

- jQuery version: 2.2.4
- bootstrap-fileinput version: v4.5.0 

yii Version 2.0.11.2

other versions:
`kartik-v/bootstrap-fileinput        v4.5.0 `
`kartik-v/bootstrap-star-rating      v4.0.1`
`kartik-v/dependent-dropdown         v1.4.4`
`kartik-v/yii2-date-range            1.6.8`
`kartik-v/yii2-dialog                v1.0.1 `
`kartik-v/yii2-editable              v1.7.6`
`kartik-v/yii2-grid                  v3.1.7`
`kartik-v/yii2-krajee-base           v1.9.4`
`kartik-v/yii2-popover-x             v1.3.4`
`kartik-v/yii2-widget-activeform     v1.4.8`
`kartik-v/yii2-widget-affix          v1.0.0`
`kartik-v/yii2-widget-alert          v1.1.0`
`kartik-v/yii2-widget-colorinput     v1.0.3`
`kartik-v/yii2-widget-datepicker     v1.4.2`
`kartik-v/yii2-widget-datetimepicker v1.4.3`
`kartik-v/yii2-widget-depdrop        v1.0.4`
`kartik-v/yii2-widget-fileinput      v1.0.5`
`kartik-v/yii2-widget-growl          v1.1.1`
`kartik-v/yii2-widget-rangeinput     v1.0.1`
`kartik-v/yii2-widget-rating         v1.0.2`
`kartik-v/yii2-widget-select2        v2.1.3`
`kartik-v/yii2-widget-sidenav        v1.0.0`
`kartik-v/yii2-widget-spinner        v1.0.0`
`kartik-v/yii2-widget-switchinput    v1.3.1`
`kartik-v/yii2-widget-timepicker     v1.0.3`
`kartik-v/yii2-widget-touchspin      v1.2.1`
`kartik-v/yii2-widget-typeahead      v1.0.1`
`kartik-v/yii2-widgets               v3.4.0`

## Isolating the problem

- [ ] This bug happens [on the plugin demos page](http://plugins.krajee.com/file-input/demo)
- [x] The bug happens consistently across all tested browsers
- [x] This bug happens when using bootstrap-fileinput without other plugins
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

**Comment by kartik-v** [comment]
Posted in the wrong repo. 

However the issue you are facing is because you have not updated packages via composer to get the latest versions of all the repos with related dependencies.

**Comment by tholag** [comment]
Thanks and sorry for getting the Repo wrong... so for anyone who faces the same problem:

It seems composer downloads a cached/older source of `vendor/yii2-widget-fileinput/src/FileInput.php` when upgrading to package version 1.0.7.

"Fix #140: Correct trait usage" seems not to be included (yet) in "https://github.com/kartik-v/yii2-widget-fileinput.git".

I cleared and checked the composer cache, but the file still contained the line removed in this fix:

```
class FileInput extends InputWidget
{
    use TranslationTrait; 
    ...
```

I downloaded from github and replaced manually, that's it.

**Comment by abude** [comment]
Hello @kartik-v 

The issue is still there.
Using composer on install or update doesn't solve the issue at all.

Thanks to @tholag and his last comment, i've copied manually from git directly, but this is not an ok solution for a product release purpose.

Do you have any stable solution for this?

Kind regards!

**Comment by kartik-v** [comment]
@abude the issue is more to do with composer settings in your app.

1. Check  your yii2 application's composer json definitions and dependencies. Are you setting extensions with conflicting or invalid dependencies that is forcing it to download older version? To test it install a fresh sample yii2 app and just select `yii2-widget-fileinput` in your composer.json require section and then install/update you should see everything works fine. 

2. you may need to see if your local composer packages are corrupted. Follow this [web tip](http://webtips.krajee.com/composer-fails-update-vendor-packages/) to see if your composer packages are corrupted. Then run a composer update it should refresh your application

**Comment by abude** [comment]
@kartik-v Thanks for the answer. :)

Actually i don't think it's a composer issue, composer will only bring the last release you have made on the repo, which doesn't include that Fix #140 if you look in your releases you will see that this fix was after the last release.

As for components from you used within the project:

- "kartik-v/yii2-widget-fileinput": "^1.0",

- "kartik-v/yii2-widget-select2": "^2.1",

- "kartik-v/yii2-mpdf": "^1.0",

- "kartik-v/yii2-datecontrol": "^1.9",

- "kartik-v/yii2-widget-datepicker": "^1.4",

Nothing outdated, right ?

Kind regards!

**Comment by kartik-v** [comment]
Try updating @abude to see if you have latest release of yii2-widget-fileinput (it is v1.0.8)

Any other issues on that repo - please post the issue in the right respective repo. This is a wrong repo to be posting the issue.

**Comment by abude** [comment]
As un update for anyone that will be looking for solution on that matter:
@kartik-v has released a new release yesterday and this issue has been solved within Fix #140 included in that release: https://github.com/kartik-v/yii2-widget-fileinput/releases/tag/v1.0.8

Thanks @kartik-v !

**Comment by tebazil** [comment]
Note: Had this problem with `kartik-v/yii2-widget-colorinput` . Updating to 1.0.5 did help.
